oecophylla/util/parse.py

```diff
@@ -0,0 +1,149 @@
+import re
+import pandas as pd
+
+# Option 1 : parse samples from directory
+
+def parse_filenames_to_df(filepaths, pattern = None, column_names = None):
+    """ Generates sample sheet from filenames using a regex pattern.
+
+    Parameters
+    ----------
+    filepaths: list of str
+        List of file paths to parse
+    pattern: str
+        Regex pattern to recognize the filenames, to load into the
+        sample sheet.
+        default: '^((.+?)_(S\d+)_(L\d+)_(R[12])_(\d+)\.(.+))$'
+    column_names: list
+        Column names for the generated sample sheet.  Note that this
+        is assumed to be in the same ordering as the regex pattern.
+        default: ['File', 'Sample', 'Index', 'Lane',
+                  'Read', 'Run', 'Extension']
+
+
+    """
+    if pattern is None:
+        pattern = '^((.+?)_(S\d+)_(L\d+)_(R[12])_(\d+)\.(.+))$'
+    if column_names is None:
+        column_names = ['File', 'Sample', 'Index', 'Lane',
+                         'Read', 'Run', 'Extension']
+    p = re.compile(pattern)
+    df = pd.DataFrame(columns = column_names)
+
+    for f in filepaths:
+        m = p.match(f)
+        if m:
+            df = df.append(dict(zip(column_names, m.groups())),
+                           ignore_index = True)
+    return df
+
+
+def get_sample_reads_df(df, seq_dir):
+    """
+    Parameters
+    ----------
+    df : pd.DataFrame
+
+    seq_dir : str
+
+    """
+    sample_reads_dict = {}
+
+    samples = list(set(df['Sample']))
+
+    for s in samples:
+        f_fps = sorted(list(df.loc[(df['Sample'] == s) & (df['Read'] == 'R1'),'File'].values))
```

Comment:
might be a bit more readable with a `for sample, sample_grp in samples.groupby('Sample'): ...`

Reply:
I like it!